addons/mail/static/src/core/common/chat_window.js

```diff
@@ -139,6 +140,27 @@ export class ChatWindow extends Component {
         }
     }
 
+    onClickContent(ev) {
+        if (!this.ui.isSmall) {
+            return;
+        }
+        const linkElement = ev.target.closest("a");
+        if (!linkElement?.href || linkElement.href.startsWith("#")) {
+            return;
+        }
+        try {
+            const url = new URL(linkElement.href);
+            if (
+                browser.location.host === url.host &&
+                browser.location.pathname.startsWith("/odoo")
+            ) {
+                this.toggleFold();
+            }
+        } catch {
+            return;
+        }
```

Comment:
Why the try/catch?
If this is relevant (I don't think so), at least it should be around the exact LOCs, and should catch exact error. Then the silent return should probably require a comment if that's not obvious enough why this is ok.

Right now the whole `this.toggleFold()` is in silent `try/catch`, which means programming errors are silent crashes, and this can lead to corrumpting the store, especially when the feature to prevent store corruption is 18.2+ at the writing of this comment: https://github.com/odoo/odoo/issues/205216

Reply:
It's in case that `new URL()` throws an error, but as you mentioned I have to move `toggleFold()` outside the try/catch to not silence its errors.